### ISSUE TITLE
server implementations should be able to specify a root context other than `context.Background()`

### ISSUE THREAD

**Issue by hdonnay** [capnproto/go-capnp#577]
While looking at adding a capnproto-rpc interface to a project, I came to the conclusion that there's no equivalent to [`http.Server`](https://pkg.go.dev/net/http#Server)'s `BaseContext`/`ConnContext`. This makes context propagation a chore, as a parent context would need to be passed in whatever concrete type is being used and then every method would need to use some tricky code to tie the signalling between the parent and request contexts together.

**Comment by matheusd** [comment]
Are you evaluating master?

**Comment by hdonnay** [comment]
Yeah; as best I can tell `Serve` ends up [here](https://github.com/capnproto/go-capnp/blob/main/rpc%2Frpc.go#L260), which unconditionally uses a background context. The context passed into `ListenAndServe` is only used for cancellation.

**Comment by hdonnay** [comment]
Yeah, that looks right.

**Comment by homier** [comment]
I think the issue could be closed since https://github.com/capnproto/go-capnp/pull/595 is merged now. @hdonnay
